src/controllers/hid/hidcontroller.cpp

```diff
@@ -104,15 +99,52 @@ int HidController::open() {
         return -1;
     }
 
-    // This isn't strictly necessary but is good practice.
-    for (int i = 0; i < kNumBuffers; i++) {
-        memset(m_pPollData[i], 0, kBufferSize);
-    }
-    m_lastPollSize = 0;
-
     setOpen(true);
     startEngine();
 
+    if (m_pHidIo != nullptr) {
+        qWarning() << "HidIo already present for" << getName();
+    } else {
+        m_pHidIo = new HidIo(m_pHidDevice,
+                std::move(m_deviceInfo),
+                m_logBase,
+                m_logInput,
+                m_logOutput);
+        m_pHidIo->setObjectName(QString("HidIo %1").arg(getName()));
+
+        connect(m_pHidIo,
+                &HidIo::receive,
+                this,
+                &HidController::receive,
+                Qt::QueuedConnection);
+
+        connect(this,
+                &HidController::getInputReport,
+                m_pHidIo,
+                &HidIo::getInputReport,
+                Qt::DirectConnection); // Enforces syncronisation of mapping and IO thread
```

Comment:
Here you are accessing data managed by the I/O thread from another thread without synchronization.

These comments are wrong, nothing gets synchronized by using Qt::DirectConnection. You are only pretending  that this direct access is safe while it is not.